File: Offer/src/linkedList/Offer06.java
package linkedList;

import java.util.LinkedList;

public class Offer06 {
    public int[] reversePrint(ListNode head) {
        LinkedList<Integer> stack = new LinkedList<>();
        while(head != null){
            stack.addFirst(head.val);
            head = head .next;
        }
        int[]res = new int[stack.size()];
        for(int i = 0; i < res.length; i++){
            res[i] = stack.removeFirst();
        }
        return res;
    }
}

File: Offer/golang/linkedList/Offer06.go
package linkedList

func reversePrint(head *ListNode) []int {
	arr := make([]int, 0)
	for head != nil {
		arr = append(arr, head.Val)
		head = head.Next
	}
	res := reverse(arr)
	return res
}

func reverse(arr []int) []int {
	for i, j := 0, len(arr)-1; i < j; i, j = i+1, j-1 {
		arr[i], arr[j] = arr[j], arr[i]
	}
	return arr
}
